src/client/producer.rs

```diff
@@ -307,12 +307,18 @@ where
     status_deagg: <A as aggregator::Aggregator>::StatusDeaggregator,
 }
 
+#[derive(Debug, Clone)]
+enum AggregatedError {
```

Comment:
This seems to just be a subset of `Error`?

Reply:
done